code/processes/tailreader.q

```diff
@@ -1,40 +1,45 @@
-\d .tr 
+\d .tr
 partitiontype:@[value;`partitiontype;`date];                               /-set type of partition (defaults to `date)
 gmttime:@[value;`gmttime;1b];                                              /-define whether the process is on gmttime or not
-determinepartition:{{@[value;`.tr.currentpartition;                        /-function to determine the partition value
-    (`date^partitiontype)$(.z.D,.z.d)gmttime]}
-  }; 
-getpartition:@[value;`getpartition;determinepartition[]];                  /-check if partition value exists and if not generate one
+determinepartition:{@[value;`.tr.currentpartition;                         /-function to determine the partition value
+  (`date^partitiontype)$(.z.D,.z.d)gmttime]
+  };
+getpartition:@[value;`getpartition;determinepartition[]];
```

Comment:
is both `determinepartition` and `getpartition` necessary? is something like this not more readble?
```
getpartition:{@[value;`getpartition;                       /-function to determine the partition value
    {(`date^partitiontype)$(.z.D,.z.d)gmttime]}
  };
currentpartition:@[value;`currentpartition;getpartition[]] 
```